# Pre-operative Imaging Concordance Analysis

## Concordance Analysis

Imports 

In [11]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

Reading in the Raw Data

In [ ]:
df = pd.read_excel('data/raw_imaging_data.xlsx')

df.head()
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 405 entries, 0 to 404
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Date of Procedure  405 non-null    datetime64[ns]
 1   Patient ID         405 non-null    int64         
 2   Age                403 non-null    object        
 3   Gender             405 non-null    object        
 4   Surgical Findings  405 non-null    object        
 5   Surgical Cure      405 non-null    object        
 6   SPECT/CT           230 non-null    object        
 7   Ultrasound         145 non-null    object        
 8   4D CT Scan         228 non-null    object        
 9   Sestamibi          174 non-null    object        
 10  MRI                15 non-null     object        
dtypes: datetime64[ns](1), int64(1), object(9)
memory usage: 34.9+ KB


Date of Procedure     Patient ID
count                            405     405.000000
mean   2020-05-21 18:36:26.666666752  243753.325926
min              2012-04-24 00:00:00  150174.000000
25%              2017-11-08 00:00:00  209246.000000
50%              2021-03-04 00:00:00  251240.000000
75%              2023-04-06 00:00:00  281328.000000
max              2025-05-13 00:00:00  312172.000000
std                              NaN   44989.927258

### Exploratory Data Analysis

#### **Dataset Size & Coverage**
- We've got **405 procedures** to work with
- The data spans from **2012 to 2025**
- Most procedures happened around **2021**


#### **Quick Observations**
- No missing data in our main columns (Date of Procedure, Patient ID) 
- Pretty even distribution of patient IDs suggests good coverage across different patients
- The date column is already in a datetime format
- 

In [ ]:
df.isna().sum()

# This shows us that we're missing two entries in the age columns

# The date column is already in a datetime format




Date of Procedure      0
Patient ID             0
Age                    2
Gender                 0
Surgical Findings      0
Surgical Cure          0
SPECT/CT             175
Ultrasound           260
4D CT Scan           177
Sestamibi            231
MRI                  390
dtype: int64